test/build/50_ch-image.bats

```diff
@@ -188,83 +188,52 @@ EOF
 }
 
 @test 'ch-image list' {
+
+    # list all images
     run ch-image list
     echo "$output"
     [[ $status -eq 0 ]]
     [[ $output = *"00_tiny"* ]]
 
-    ### IMAGE_REF ###
-
     # does not exist remotely
-    run ch-image list foo:bar
+    run ch-image list doesnotexist:latest
+    echo "$output"
     [[ $status -eq 1 ]]
-    [[ $output = *'error'* ]]
-    [[ $output = *'GET failed'* ]]
-    [[ $output = *'expected status {200, 403} but got 400: Bad Request'* ]]
+    [[ $output = *'GET failed; expected status {200, 403} but got 400'* ]]
 
-    # in storage, does not exist remotely, no fat manifest
+    # in storage, does not exist remotely
     run ch-image list 00_tiny
     echo "$output"
     [[ $status -eq 1 ]]
-    [[ $output = *'in local storage:    yes'* ]]
-    [[ $output = *'{200, 403} but got 400: Bad Request'* ]]
+    [[ $output = *'in local storage:'*'yes'* ]]
+    [[ $output = *'GET failed; expected status {200, 403} but got 400'* ]]
 
-    # exists remotely, fat manifest exists
-    run ch-image list debian:buster
+    # not in storage, exists remotely, fat manifest exists
+    run ch-image list debian:buster-slim
     echo "$output"
     [[ $status -eq 0 ]]
-    [[ $output = *'available remotely:  yes'* ]]
-    [[ $output = *'in local storage:    no'* ]]
-    [[ $output = *'remote arch-aware:   yes'* ]]
-    [[ $output = *'archs available:'* ]]
-    [[ $output = *'386 amd64 arm/v5 arm/v7 arm64/v8 mips64le ppc64le s390x'* ]]
-
-    # exists remotely, no fat manifest
-    run ch-image list charliecloud/metadata:2021-01-15 --no-cache
+    [[ $output = *'available remotely:'*'yes'* ]]
+    [[ $output = *'in local storage:'*'no'* ]]
+    [[ $output = *'remote arch-aware:'*'yes'* ]]
```

Comment:
The reason I preserved the whitespace here is to avoid false positive results. When the ch-image list output is something like below:

```
available remotely:  no
in local storage:    yes
remote arch-aware:   no
```

Both the test cases `[[ $output = *'available remotely:'*'yes'* ]]` and `[[ $output = *'in local storage:'*'no'* ]]` will pass even though they shouldn't.

Reply:
Oh, good catch. I was thinking to avoid false negatives if the number of spaces changes. Options include: (1) revert, and deal with false negatives if/when they arise, (2) change to regular expression with variable number of spaces. Option 2 is rather messy because it's hard to write Bash regexes as a literal (see examples elsewhere; we universally set a variable `$re`. We could also go with Option 1 and then change it to Option 2 if it becomes a problem.

Reply:
I prefer going with option one (1) for now. If (when) it becomes an issue we can explore option 2 thoroughly.